src/v2i-hub/ImmediateForwardPlugin/src/ImmediateForwardPlugin.h

```diff
@@ -62,10 +62,12 @@ class ImmediateForwardPlugin : public tmx::utils::PluginClient
 	std::array<svr_list, 4> _udpClientList;
 	std::vector<MessageConfig> _messageConfigMap;
 	std::map<std::string, int> _messageCountMap;
-	std::string signatureData; 
-	std::string url; 
-	std::string baseurl; 
-	unsigned int signState; 
+	std::string signatureData;
+	std::string url;
+	std::string baseurl;
+	std::string txMode;
+	unsigned int signState;
```

Comment:
Does we still need any signState if replaced by enableHSM?

Reply:
In most cases, we don't sign the message and an RSU has to sign it. The RSU will expect to receive the Active Message file with this field set to True. So, this allows us to select True without trying to sign the message.

Reply:
- Is this Active Message file field set by IMF enableHSM? Or ? 
- Are you saying that we enableHSM = true without sign the message and the signState could be 1?
I am not sure I understand the concept of the enableHSM and signState and if there are any relationships between them. I see this comment before the if (enableHSM=true): `/// if signing is Enabled, request signing with HSM`. It sounds like there are some relationship between them?

Reply:
The field in the Active Message file is set by signState.
If you want V2X Hub to sign messages, you have to set enableHSM to 1. You should also set signState to 1 if your RSU is expecting to receive a signed message or will be signing a message. In some cases, you'll want to set both to 1, but in most cases, you'll only want to set signState to 1.

Reply:
If we are in most cases and we have enabledHSM to false, signState = 1, is this signState field value 1 updated in the Active Message File? 

Is this code here populate the active message file or?
`
os << "Signature=" << (signState == 1 ? "True" : "False") << "\n" << "Encryption=False\n";
`

Reply:
Yep! That's where it's updated

Reply:
Thanks! but if enabledHSM = false, this Signature is not populated.

Reply:
The entry for Signature is independent from enabledHSM. It's outside of that if-else case and only relies on signState.